# -----------------------------------------------------------------------------
#                        SLAM system definition for MOLA
#
# This file just replays (no SLAM) the raw data of a sequence of the KAIST dataset
# -----------------------------------------------------------------------------

modules:
  # =====================
  # KaistDataset
  # =====================
  - type: mola::KaistDataset
    name: kaist_input
    execution_rate: 50 # Hz
    #verbosity_level: INFO
    #export_to_rawlog: kaist_${KAIST_SEQ}.rawlog
    gui_preview_sensors:
      - raw_sensor_label: VLP_left
        decimation: 1
        win_pos: 5 5 400 400
      - raw_sensor_label: VLP_right
        decimation: 1
        win_pos: 415 5 400 400
    params:
      base_dir: ${KAIST_BASE_DIR}
      sequence: ${KAIST_SEQ}
      time_warp_scale: 2.0
      publish_odometry: true
      publish_VLP_left: true
      publish_VLP_right: true
  # =====================
  # MolaViz
  # =====================
  - name: viz
    type: mola::MolaViz
    #verbosity_level: DEBUG
    params: ~ # none
